### ISSUE TITLE
Multi Instance Call Activities stopped working since 1.1.1

### ISSUE THREAD

**Issue by kaihasse** [sartography/SpiffWorkflow#151]
### Problem:
Multi Instance Call Activities stopped working since Version 1.1.1.

### Reason:
Class determination in method **_handle_special_cases()** works different than **isinstance()** so _CallActivity_ can't be identified as subclass of _SubWorkflowTask_.

### Fix:
In bpmn/specs/MultiInstanceTask.py:

```
425c425
<         classes = [BusinessRuleTask,ScriptTask,SubWorkflowTask,SubWorkflow]
---
>         classes = [BusinessRuleTask,ScriptTask,SubWorkflowTask,SubWorkflow,CallActivity]
```

**Comment by danfunk** [comment]
Blast it.  Thank you for letting me know.  Clearly need a better test around this.  Will let you know when it is corrected.

**Comment by danfunk** [comment]
@kaihasse  - would love to know a little about how you are using multiprocess with call activities.  Perhaps we can model our test around your specific use case.

**Comment by kaihasse** [comment]
At the moment I am just testing the features of your library. I was thinking about the following scenario: 
For a final product, a production workflow is to be triggered for all elements of a bill of materials and merged into a complete product at the end. The production of the individual components takes place in parallel. 

I attached a small test workflow showing the problem. I also didn't figure out how to update the main workflow data from the subworkflows...

[call_activity_multi.zip](https://github.com/sartography/SpiffWorkflow/files/7861798/call_activity_multi.zip)

**Comment by danfunk** [comment]
We will take a look at the diagram and think through your use case.

 re:  "I also didn't figure out how to update the main workflow data from the subworkflows..."
Data is passed from one task to the next in the task.data.  When a task completes, all of the children of that task will inherit it's data.  This is true even when the child is in a sub-workflow, and it is true when the child of a sub-workflow task is a task within the main workflow.  We are actively working on better ways to manage data through the DataObject and DataStore concepts in BPMN.  But the spec doesn't offer a ton of direction on how to manage data.  We'd very much like to introduce some concept of scope and create a sense of scope between tasks, sub-processes, and the main workflow.  Hopefully we will have more clarity on this next week.  

Thank you for the example.  Will get this rolled in today.

**Comment by danfunk** [comment]
No wonder you didn't see how the data worked between a workflow and a subworkflow, as in your example test case, it was badly broken.  I've got a pull request out now that should fix it, and we can make this a part of the release for 1.1.2  

I created this [Multi-Instance SubWorkflow BPMN](https://github.com/sartography/SpiffWorkflow/blob/96b3f02347661a324ff4cafc308cb645c73edcd1/tests/SpiffWorkflow/bpmn/data/sub_workflow_multi.bpmn) as our test case. This BPMN will result in a final data structure, as asserted in [this test](https://github.com/sartography/SpiffWorkflow/blob/96b3f02347661a324ff4cafc308cb645c73edcd1/tests/SpiffWorkflow/bpmn/SubWorkflowMultiTest.py).
```
{
     'a': {'name': 'Apple',
           'new_info': 'Adding this!'},
     'b': {'name': 'Bubble',
           'new_info': 'Adding this!'},
     'c': {'name': 'Crap, I should write better code',
           'new_info': 'Adding this!'}})
} 
```

**Comment by kaihasse** [comment]
Thanks for the quick fix and the working examples! 

Unfortunately there still seems to be a problem with parallel multi instances: Updating data works fine for sequential multi instances but for parallel ones it seems that **only the last element** of the collection is updated.

**Comment by danfunk** [comment]
This code could use some refactoring.  I was a little simplemineded in the last fix, this will allow the multi-instance data to function when returning from a parallel sub-workflow.   Thank you for sticking with me @kaihasse
https://github.com/sartography/SpiffWorkflow/pull/153

**Comment by kaihasse** [comment]
Hello Dan,
I am sorry to say that there still seems to be an error...
Adding new data from subworkflows works fine but there is a problem with updating existing fields.

I attached a bunch of tests to show the problem:
[tests.zip](https://github.com/sartography/SpiffWorkflow/files/7881752/tests.zip)

**Comment by danfunk** [comment]
Tricky Tricky,  How about this pull request:  https://github.com/sartography/SpiffWorkflow/pull/155
I was able to run your all of your tests successfully with these latest changes.

**Comment by kaihasse** [comment]
@danfunk 
Looks good to me! Thanks again for the quick fix...
